nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/StandardOauth2AccessTokenProvider.java

```diff
@@ -236,6 +252,19 @@ protected Collection<ValidationResult> customValidate(ValidationContext validati
                 .build());
         }
 
+        String clientAuthenticationStrategy = validationContext.getProperty(CLIENT_AUTHENTICATION_STRATEGY).getValue();
+        try {
+            ClientAuthenticationStrategy.valueOf(clientAuthenticationStrategy);
+        } catch (IllegalArgumentException ex) {
+            validationResults.add(new ValidationResult.Builder().subject(CLIENT_AUTHENTICATION_STRATEGY.getDisplayName())
+                .valid(false)
+                .explanation(String.format("Unrecognized %s value '%s'",
+                    CLIENT_AUTHENTICATION_STRATEGY.getDisplayName(),
+                    clientAuthenticationStrategy)
+                )
+                .build());
+        }
```

Comment:
This custom validation is not necessary because the allowable values will prevent unsupported values from being configured.
```suggestion
```